### PR TITLE
Replace Linkify by Url highlight

### PR DESCRIPTION
**Pull request by vstelmakh**
Suggestion to replace Linkify by [Url highlight](https://github.com/vstelmakh/url-highlight) library. It covers more url cases and urls without scheme. See comparison table bellow.

### `Linkify` vs `Url highlight` comparison
|                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | Linkify                                                                  | Url highlight                                                |
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|--------------------------------------------------------------|
| Supported schemes                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | ![](https://via.placeholder.com/10/d8c254/000000?text=+) http(s), ftp(s) | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Any |
| Simple url <br> <details>   <summary>Details</summary>   Simple url with scheme: `http://example.com`   <ul>     <li>Linkify: `<a href="http://example.com">http://example.com</a>`</li>     <li>Url highlight: `<a href="http://example.com">http://example.com</a>`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes             | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |
| No scheme <br> <details>   <summary>Details</summary>   Valid domain with no scheme: `example.com`   <ul>       <li>Linkify: `example.com`</li>     <li>Url highlight: `<a href="http://example.com">example.com</a>`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | ![](https://via.placeholder.com/10/f03c15/000000?text=+) No              | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |
| No scheme with path <br> <details>   <summary>Details</summary>   Valid domain with no scheme with path: `example.com/path`   <ul>     <li>Linkify: `<a href="http://example.com/path">example.com/path</a>`</li>     <li>Url highlight: `<a href="http://example.com/path">example.com/path</a>`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes             | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |
| Invalid domain <br> <details>   <summary>Details</summary>   Not existing domain but string looks like url: `example.txt/path`   <ul>     <li>Linkify: `<a href="http://example.txt/path">example.txt/path</a>`</li>     <li>Url highlight: `example.txt/path`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | ![](https://via.placeholder.com/10/f03c15/000000?text=+) Yes             | ![](https://via.placeholder.com/10/439e0a/000000?text=+) No  |
| Email <br> <details>   <summary>Details</summary>   Generic email: `user@example.com`   <ul>     <li>Linkify: `<a href="mailto:user@example.com">user@example.com</a>`</li>     <li>Url highlight: `<a href="mailto:user@example.com">user@example.com</a>`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes             | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |
| Complex url <br><details>   <summary>Details</summary>   Urls could contain lots of special chars and brackets nesting. Example: `http://elk.example.com/app/kibana#/discover?_g=()&_a=(columns:!(_source),index:'deve-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'*')),sort:!('@timestamp',desc))`   <ul>     <li>Linkify: `<a href="http://elk.example.com/app/kibana#/discover?_g=()&_a=">http://elk.example.com/app/kibana#/discover?_g=()&_a=</a>(columns:!(_source),index:'deve-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'*')),sort:!('@timestamp',desc))`</li>     <li>Url highlight: `<a href="http://elk.example.com/app/kibana#/discover?_g=()&_a=(columns:!(_source),index:'deve-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'*')),sort:!('@timestamp',desc))">http://elk.example.com/app/kibana#/discover?_g=()&_a=(columns:!(_source),index:'deve-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'*')),sort:!('@timestamp',desc))</a>`</li>   </ul> </details> | ![](https://via.placeholder.com/10/f03c15/000000?text=+) No              | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |
| Url with userinfo <br> <details>   <summary>Details</summary>   Url could contain basic auth credentials: `http://user:password@example.com`   <ul>     <li>Linkify: `<a href="http://user:<a href="mailto:password@example.com">password@example.com</a>">http://user:<a href="mailto:password@example.com">password@example.com</a></a>`</li>     <li>Url highlight: `<a href="http://user:password@example.com">http://user:password@example.com</a>`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | ![](https://via.placeholder.com/10/f03c15/000000?text=+) No              | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |
| Html encoded input <br> <details>   <summary>Details</summary> Escaping html in string: `<b>http://example.com</b>` will result to: `&lt;b&gt;http://example.com&lt;/b&gt;`      <ul>     <li>Linkify: `&lt;b&gt;<a href="http://example.com&lt;/b&gt">http://example.com&lt;/b&gt</a>;`</li>     <li>Url highlight: `&lt;b&gt;<a href="http://example.com">http://example.com</a>&lt;/b&gt;`</li>   </ul> </details>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | ![](https://via.placeholder.com/10/f03c15/000000?text=+) No              | ![](https://via.placeholder.com/10/439e0a/000000?text=+) Yes |

#### Legend
 - ![](https://via.placeholder.com/10/439e0a/000000?text=+) - Proper match  
 - ![](https://via.placeholder.com/10/d8c254/000000?text=+) - Partial match  
 - ![](https://via.placeholder.com/10/f03c15/000000?text=+) - Invalid match

### CODE REVIEW

File: src/UrlHighlightFactory.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+use VStelmakh\UrlHighlight\Highlighter\HtmlHighlighter;
+use VStelmakh\UrlHighlight\UrlHighlight;
+
+class UrlHighlightFactory
+{
+    public static function createUrlHighlight(string $defaultScheme, array $attributes): UrlHighlight
+    {
+        $highlighter = new HtmlHighlighter($defaultScheme, $attributes);
+        return new UrlHighlight(null, $highlighter);
```
Comment (mnapoli) [inline src/UrlHighlightFactory.php:13]: Thanks for the PR!

The only thing I see that could be improved is removing that class and moving these 2 lines inside the factory in `config.php`, could you do that?

Thanks!

Comment (vstelmakh) [reply src/UrlHighlightFactory.php:13]: That's make sense. Will do so.

Comment (vstelmakh) [reply src/UrlHighlightFactory.php]: Resolved

**Comment by mnapoli** [comment]
That's really good, thanks a lot!

**Pull request merged**